starlark/library.go

```diff
@@ -904,21 +885,33 @@ func (r rangeValue) Len() int          { return r.len }
 func (r rangeValue) Index(i int) Value { return MakeInt(r.start + i*r.step) }
 func (r rangeValue) Iterate() Iterator { return &rangeIterator{r, 0} }
 
+// rangeLen calculates the length of a range with the provided start, stop, and step.
+// caller must ensure that step is non-zero.
```

Comment:
This precondition relies on the Sliceable.Slice interface having the same precondition, which is does implicitly, but I forgot to document it. Could you?

Reply:
Done, in an additional commit. Happy to document other places as well if you like.